ale_linters/c/gcc.vim

```diff
@@ -10,6 +10,17 @@ if !exists('g:ale_c_gcc_options')
 endif
 
 function! ale_linters#c#gcc#GetCommand(buffer) abort
+    " attempt to get args from compilation database
+    let l:args = ale#c#FindCompileArgs(a:buffer)
+
+    " if we've found compile args then just use those
+    if len(l:args) > 0
```

Comment:
It might be better to test for `has_key(l:args, 'args') && has_key(l:args, 'directory')`